core/reflection.jvm/resources/META-INF/proguard/kotlinreflect.pro

```diff
@@ -0,0 +1,14 @@
+##---------------Begin: Kotlin-reflect ----------
+# Keep Metadata annotations so they can be parsed at runtime.
+-keep class kotlin.Metadata { *; }
```

Comment:
I wonder this is not actually needed. Recently, I started working on synthesizing @Metadata from scratch (for R8-processed Kotlin library), and kept adding more tests. During that, I found that, even without this rule, R8 will still keep metadata (annotation) associated to classes.

More conversion/details are here: https://r8-review.googlesource.com/c/r8/+/46240/1/src/test/java/com/android/tools/r8/kotlin/metadata/MetadataRenameInParametertypeTest.java#70

Reply:
Ah, one caveat is, though, R8 with this fix (https://r8-review.googlesource.com/c/r8/+/44220) will keep metadata annotation without definition. For prior versions of R8, it may be the case that we still need this rule.

Reply:
Yeah I think it's best to keep this as a conservative rule. We've had [reports in Moshi ](https://github.com/square/moshi/pull/1038) that Metadata is stripped in android builds without a custom keep rule for some users, though unsure what version they're using or if they're just using proguard.

Reply:
@jsjeon what version of R8 would that fix be in? I'm not sure how to check the latest release history or what's in them off-hand there

Reply:
Ah, it's even difficult for me to check which version includes it. :\ Based on the timeline, I guess AGP 3.6 will include that.

However, that fix I mentioned was about keeping annotations associated with classes/members _if_ the definition of annotations is missing. Therefore, if annotation class `Metadata` is given (via kotlin reflect artifact jar of course), i.e., not missing, it is still the case that R8 will take it away if it thinks it's not used.

------

I was about to mention the case like https://issuetracker.google.com/issues/146534384 where `kotlin.Metadata` itself is retrieved at runtime. In that case, yeah, this rule is necessary.

Reply:
Sounds like it's safest to just keep this for now